action.yaml

```diff
@@ -15,10 +15,31 @@ inputs:
     description: 'Used to specify whether caching is needed. Set to false, if you would like to disable caching.'
     required: false
     default: 'false'
+  token:
+    description: >
+      Access token used to check out the Trivy repository.
+      The token is required when using GitHub Enterprise Server (GHES).
+      https://github.com/actions/create-github-app-token can be used to obtain such a token.
+      The token should be limited to read access only for public repositories.
+      See more details in https://github.com/aquasecurity/setup-trivy/issues/10
+    required: false
+    default: ''
 
 runs:
   using: 'composite'
   steps:
+    - name: GitHub token
+      id: github-token
+      shell: bash
+      ## ${{ github.token }} is default value for actions/checkout
+      ## cf. https://github.com/actions/checkout/blob/eef61447b9ff4aafe5dcd4e0bbf5d482be7e7871/action.yml#L24
+      run: |
+        if [ -z "${{ inputs.token }}" ]; then
+          echo "token=${{ github.token }}" >> $GITHUB_OUTPUT
+        else
+          echo "token=${{ inputs.token }}" >> $GITHUB_OUTPUT
+        fi
```

Comment:
Do we still need this step if we set the default value of `token` like [actions/checkout](https://github.com/actions/checkout/blob/163217dfcd28294438ea1c1c149cfaf66eec283e/action.yml#L24)? `inputs.token` points to the default GitHub token if `token` is not specified.

Reply:
initially, I used `${{ github.token }}` as default value for `inputs.token`.

But `actions/checkout` doesn't work if `token` is empty (but works if `token` line is missing) - https://github.com/DmitriyLewen/test-trivy-action/actions/runs/11476162864/job/31935565983

So, there are 2 ways to avoid errors in `trivy-action`:
- use `${{ github.token }}` as default value for `token-setup-trivy` in `trivy-action`.
- add step in `setup-trivy` (my way).

I thought it's better to add step in `setup-trivy` to avoid questions in `trivy-action` about using `${{ github.token }}`. 
But if you want, I can use the 1st method.

Example:
setup-trivy (`${{ github.token }}` is default value for `inputs.token`) - https://github.com/DmitriyLewen/setup-trivy/blob/0faeae687c92b01123d6eaafbe78091bc8b610eb/action.yaml
trivy-action (`` is  default value for `inputs.token-setup-trivy`)- https://github.com/DmitriyLewen/trivy-action/blob/e78daca2a1cb92de6fd1ba645bd146b851efc10c/action.yaml
test run - https://github.com/DmitriyLewen/test-trivy-action/actions/runs/11476162864/job/31935565983

Reply:
>initially, I used ${{ github.token }} as default value for inputs.token.

What changed your mind to go for the second approach (additional step)? Is there any benefit?

>But actions/checkout doesn't work if token is empty (but works if token is specified) - https://github.com/DmitriyLewen/test-trivy-action/actions/runs/11476162864/job/31935565983

I'm probably missing something, but my suggestion will not fall into this case, right?

Reply:
> What changed your mind to go for the second approach (additional step)? Is there any benefit?

I didn't think `actions/checkout` uses `token` for public repos.
But IIUC `actions/checkout` uses same logic as `ghcr` (you need to use a valid token even if you pull a public image).

That's why I thought `trivy-action` users would have questions about `github.token`, how we use it, etc.

So the only benefit would be fewer questions (if any)

> I'm probably missing something, but my suggestion will not fall into this case, right?

Since we are using a chain of actions (`trivy-action` -> `setup-trivy` -> `actions/checkout`), we need to set a default value (`github.token`) for `trivy-action` and `setup-trivy`.
If we use an empty default value for `trivy-action` - `setup-trivy` will get `` instead of `nil` and we will get an error.

I tried removing `default` line (I was hoping that would give me `nil`), but I got the same behavior.

Reply:
I now realized that another plus is that we don't have to set the default value twice. 
If there is a need to change it, we will do it in one place

Reply:
>Since we are using a chain of actions (trivy-action -> setup-trivy -> actions/checkout), we need to set a default value (github.token) for trivy-action and setup-trivy.

Yes, I suggested adding `default: ${{ github.token }}`. It wouldn't be empty, right? What's wrong?

Reply:
> > Since we are using a chain of actions (trivy-action -> setup-trivy -> actions/checkout), we need to set a default value (github.token) for trivy-action and setup-trivy.
> 
> Yes, I suggested adding `default: ${{ github.token }}`. It wouldn't be empty, right? What's wrong?

Is it because the `default: ${{ github.token }}` doesn't propagate down the chain of actions? If it does, I don't see why we shouldn't use it as the default.

Reply:
You are right. It seems I have confused myself.

Reply:
I updated PR.